File: sieve_of_eratosthenes.py
# Implement sieve of eratosthenes
def primes(x):
    assert (x >= 0)
    # Make sure x > 0 so that the function doesnt return any number less than 0
    sieve = [1 for v in range(x + 1)]
    # prepare sieve such that it contains all the numbers less than x
    primes = []
    # Initialize a list of Primes
    for i in range(2, x + 1):
        if sieve[i] == 1:
            primes.append(i)
            for j in range(i, x + 1, i):
                sieve[j] = 0
    return primes


# Driver Function
if __name__ == '__main__':
    x = int(input("enter the number until which you want to find the prime number "))
    print (primes(x))

File: sieveoferatosthenes.go
package main
import "fmt"
func main(){
	var n int
	i:=5
	c:=2
	var f int
	fmt.Printf("Enter the range of prime number required: ")
	fmt.Scanf("%d",&n)
	array:=make([]int,n)
	array[0]=2
	array[1]=3
	for i <= n{
		f=1
		for j:=0;j<c&&array[j]*array[j]<=i;j++{
			if i%array[j]==0{
				f=0
				break
			}
		}
		if f>0{
			array[c]=i
			c++
		}
		i++
	}
	fmt.Printf("Prime number upto %d are: ",n)
	for i:=0;i<c;i++{
		fmt.Print(array[i])
		fmt.Print(" ")
	}
	fmt.Printf("\nTotal Prime numbers are %d: ",c)
}
